WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift

```diff
@@ -66,16 +64,44 @@ extension NewOrderHostingController: UIAdaptivePresentationControllerDelegate {
     }
 
     func presentationControllerDidAttemptToDismiss(_ presentationController: UIPresentationController) {
-        UIAlertController.presentDiscardChangesActionSheet(viewController: self, onDiscard: { [weak self] in
-            self?.viewModel.discardOrder()
-            self?.dismiss(animated: true, completion: nil)
-        })
+        presentDiscardChangesActionSheet(onDiscard: discardOrderAndDismiss)
+    }
+}
+
+/// Private methods
+///
+private extension NewOrderHostingController {
+    func checkUnsavedChangesAndDismissIfPossible() {
+        guard !viewModel.hasChanges else {
+            presentDiscardChangesActionSheet(onDiscard: discardOrderAndDismiss)
+            return
+        }
+
+        self.dismiss(animated: true)
+    }
+
+    func presentDiscardChangesActionSheet(onDiscard: (() -> Void)?) {
```

Comment:
It's a good question 😀 The question for me was why we need to escape this closure if our `UIAlertController`doesn't do it 🤔 And I couldn’t find any good reason. However, after knowing the feature flag, the rules changes, so I think this method will disappear 🗑️

Reply:
> The question for me was why we need to escape this closure if our UIAlertControllerdoesn't do it

I just realized that the `onDiscard` closure of the `UIAlertController` extension method is optional too, that is why it's escaping. It's weird that the closure needs to be optional since we're more likely to trigger some action when discarding changes 🤷‍♀️ .

Reply:
Yep, it's weird 😅 In my opinion, if the view is destroyed there is no point in keeping the closures, however, I don't see any downside if we do 😅 So what do you think, should we escape them?

Reply:
I believe the closure needs to be escaped if it's not optional, since it will be called as part of an `UIAlertAction`. So it's not about whether it is necessary to keep the closure, WDYT?

Reply:
ok, it's a good point, I'm going to make the change 👍🏼 
Thanks! 🙂